pkg/kube/inject/template.go

```diff
@@ -195,12 +196,25 @@ func appendMultusNetwork(existingValue, istioCniNetwork string) string {
 			log.Warnf("Unable to unmarshal Multus Network annotation JSON value: %v", err)
 			return existingValue
 		}
+		istioCniNetworkNs, istioCniNetworkName := kstrings.SplitQualifiedName(istioCniNetwork)
 		for _, net := range networks {
-			if net["name"] == istioCniNetwork {
-				return existingValue
+			if net["name"] == istioCniNetworkName {
+				ns := net["namespace"]
```

Comment:
nit:

this reads a little easier if the comparison is `if ns["namespace"] == istioCniNetworkNs`

If the `nil` check is critical, then `if ns := ns["namespace"]; ns != nil && ns == istioCniNetworkNs`